packages/aws-cdk-lib/aws-cloudfront/lib/distribution.ts

```diff
@@ -724,7 +728,10 @@ export class Distribution extends Resource implements IDistribution {
 
   private renderCacheBehaviors(): CfnDistribution.CacheBehaviorProperty[] | undefined {
     if (this.additionalBehaviors.length === 0) { return undefined; }
-    return this.additionalBehaviors.map(behavior => behavior._renderBehavior());
+    return this.additionalBehaviors.map(behavior => {
+      this.validateGrpc(behavior);
+      return behavior._renderBehavior();
+    });
```

Comment:
If even one behavior has gRpc, an error will occur, so check each behavior.

The example is the case for gRPC with a httpVersion not including HTTP2:

```ts
new cloudfront.Distribution(stack, 'TestDistribution', {
  httpVersion: cloudfront.HttpVersion.HTTP3,
  defaultBehavior: {
    origin,
    allowedMethods: cloudfront.AllowedMethods.ALLOW_ALL,
    // enableGrpc: true,
  },
  additionalBehaviors: {
    '/second': {
      origin,
      allowedMethods: cloudfront.AllowedMethods.ALLOW_ALL,
      // enableGrpc: true,
    },
    '/third': {
      origin,
      allowedMethods: cloudfront.AllowedMethods.ALLOW_ALL,
      enableGrpc: true,
    },
  },
});
```